lib/scheduler.js

```diff
@@ -29,7 +29,7 @@ function Scheduler (options) {
     var self = this;
 
     self.options = {};
-    self.options.frameworkName = options.frameworkName.replace(/ /g, "-") || "mesos-framework." + uuid.v4();
+    self.options.frameworkName = (options.frameworkName || "mesos-framework." + uuid.v4()).replace(/ /g, "-");
```

Comment:
Not sure why the replace was moved. There shouldn't be any space in a UUID, but in a framework name the can potentially be some... Please change it back.